ert/data/record/_record.py

```diff
@@ -236,36 +238,43 @@ def collection_type(self) -> RecordCollectionType:
         return self._collection_type
 
 
-def path_to_bytes(file_path: pathlib.Path) -> bytes:
+async def path_to_bytes(file_path: pathlib.Path) -> bytes:
+    _executor = futures.ThreadPoolExecutor()
+    return await get_event_loop().run_in_executor(
+        _executor, _sync_path_to_bytes, file_path
+    )
+
+
+def _sync_path_to_bytes(file_path: pathlib.Path) -> bytes:
     tar_obj = io.BytesIO()
     with tarfile.open(fileobj=tar_obj, mode="w") as tar:
         tar.add(file_path, arcname="")
     return tar_obj.getvalue()
 
 
-def load_collection_from_file(
+async def load_collection_from_file(
     file_path: pathlib.Path,
     mime: str,
     ensemble_size: int = 1,
     is_directory: bool = False,
 ) -> RecordCollection:
     if mime == "application/octet-stream":
         if is_directory:
+            _data = await path_to_bytes(file_path)
```

Comment:
I don't see are reason why you should start the `_data` variable with an underscore, it is not a private variable in a class or a module.